tests/func/experiments/test_stash_exp.py

```diff
@@ -0,0 +1,70 @@
+import logging
+
+import pytest
+from funcy import first
+
+from dvc.exceptions import ReproductionError
+
+
+@pytest.mark.parametrize("tmp", [True, False])
+@pytest.mark.parametrize("staged", [True, False])
+def test_deleted(tmp_dir, scm, dvc, caplog, tmp, staged):
+    if not tmp:
+        pytest.xfail("TODO: https://github.com/iterative/dvc/issues/6297")
```

Comment:
Deleted files are still now handled properly for workspace in local `init_git`. Will fix in follow-up as the current P.R. already fixes #8998